cloudstate-kotlin-support/src/main/kotlin/io/cloudstate/kotlinsupport/transcoding/EventSourcedTranscoder.kt

```diff
@@ -0,0 +1,114 @@
+package io.cloudstate.kotlinsupport.transcoding
+
+import io.cloudstate.kotlinsupport.ReflectionHelper
+import io.cloudstate.javasupport.eventsourced.EventSourcedEntity as JEventSourcedEntity
+
+import io.cloudstate.kotlinsupport.api.eventsourced.*
+import io.cloudstate.kotlinsupport.logger
+import io.cloudstate.kotlinsupport.transcoding.eventsourced.*
+import net.bytebuddy.ByteBuddy
+import net.bytebuddy.agent.ByteBuddyAgent
+import net.bytebuddy.asm.MemberAttributeExtension
+import net.bytebuddy.dynamic.DynamicType
+import net.bytebuddy.dynamic.loading.ClassReloadingStrategy
+import net.bytebuddy.matcher.ElementMatchers
+
+class EventSourcedTranscoder(private val clazz: Class<*>): Transcoder {
+    private val log = logger()
+    private val helper = ReflectionHelper()
+
+    init {
+        log.debug("Initializing ByteBuddy Agent....")
+        ByteBuddyAgent.install()
```

Comment:
@sleipnir I have no knowledge in kotlin, so bare with my question: why would you like to have such a dependecy like bytebuddy? Is this and such an agent really needed?

Reply:
Hello @marcellanz , yes and no. Kotlin's implementation delegates much of the work to the Java library. For the support of the eventsourced and crdt dsl kotlin that will come next I will need this, but I am currently using it to transcribe the kotlin annotations to the java annotations and make everything work transparently. Could you do it any other way? Perhaps! But the execution of this is very cheap to worry about, as it is executed before any real call and only once per registered entity.

Reply:
@marcellanz Although I don't know if I understood your question correctly. Did you ask about the dependency itself or the Agent?
For the first I answered above, for the second I need the Agent because, I will transform an existing class and not create a new class and it is only allowed to do this in the jvm at runtime via Agent

Reply:
The same strategy of using java-support could be used to deploy other 'JVM support languages' like JRuby for example.

@marcellanz Let me know if you see any potential problems with this.

Reply:
>Although I don't know if I understood your question correctly. Did you ask about the dependency itself or the Agent?

@sleipnir Both. I'm a bit averse to libraries not absolutely necessary. Having an agent (even without *.c extensions, except for Windows in this example) could be a burden for someone to use it; but I might be a bit old fashioned. @viktorklang wdyt?

Reply:
> > Although I don't know if I understood your question correctly. Did you ask about the dependency itself or the Agent?
> 
> @sleipnir Both. I'm "not in favor" to libraries not absolutely necessary. Having an agent (even without *.c extensions, except for Windows in this example) could be a burden for someone to use it; but I might be a bit old fashioned. @viktorklang wdyt?
> 
> As an example, I work with some customers where I would have to explain thoroughly such a dependency.

Hi @marcellanz, thanks for the explanations.
I can only partially agree with your argument. I understand your use case but I find it very restrictive for the development of a library. Kotlin support has two important direct dependencies, the java-support library and the bytebuddy, the largest of which is the java-support library with a little more than 2mb and the bytebuddy has a little more than 250k bytes in size. That is, it is definitely not a heavy dependency. Today dependency is strictly necessary, in the future it can be replaced in other ways, but right now I am directing the energy to deliver a fluent interface to the end user. I had already commented somewhere else that I do not remember that this issue would be addressed in the future so I am opening an issue of improvements to make this registered for everyone.

Thanks Marcel :+1:

Reply:
@sleipnir Just for my understanding, what are the alternatives?

Reply:
> @sleipnir Just for my understanding, what are the alternatives?

Hi @viktorklang 
The quick alternative is to remove as well as everything else that it refers to, that is, the annotations in io.cloudstate.kotlinsupport.api and the API fluency.
The time-consuming alternative is to rewrite this without using any bytecode rewrites. I talk more about this in the issue https://github.com/cloudstateio/kotlin-support/issues/6

But I definitely don't think that is essential at the moment.

Reply:
@sleipnir Would an annotation-based API be expected for Kotlin programmers?

Reply:
> Would an annotation-based API be expected for Kotlin programmers?

I don't see why it wouldn't be, many Kotlin frameworks and libraries use them. One of the most famous, for example, Ktor uses annotations to map http endpoints to controllers.

Reply:
@sleipnir Good :) The most important thing is that it feels familiar to the Kotlin crowd. :)

Reply:
Some others examples could be found here https://kotlinlang.org/docs/tutorials/android-frameworks.html

Reply:
Ok Guys thanks for the comments

Reply:
Thank you for the hard work @sleipnir!

Are there any comments to address on this PR before it's ready for merging?

Reply:
@marcellanz anything to add?

Reply:
> @marcellanz anything to add?

@sleipnir nothing further from me :-) thanks